### PR TITLE
More fixes and enhancements for cache_bench

### PR DESCRIPTION
**Pull request by pdillinger**
Summary:
Mostly things for using cache_bench for stress/correctness testing.
* Make secondary_cache_uri option work with HCC (forgot to update when secondary support was added for HCC)
* Add -pinned_ratio option to keep more than just one entry per thread pinned. This can be important for testing eviction stress.
* Add -vary_capacity_ratio for testing dynamically changing capacity.

Also added some overrides to CacheWrapper to help with diagnostic output.

Test Plan: manual, make check

### CODE REVIEW

File: cache/cache_bench_tool.cc [hunk 1]
```diff
@@ -651,69 +675,78 @@ class CacheBench {
     uint64_t lookup_misses = 0;
     uint64_t lookup_hits = 0;
     // To hold handles for a non-trivial amount of time
-    Cache::Handle* handle = nullptr;
+    std::deque<Cache::Handle*> pinned;
+    size_t pin_count =
+        ((FLAGS_cache_size * FLAGS_pinned_ratio + FLAGS_value_bytes) /
+             FLAGS_value_bytes +
+         thread->tid) /
+        FLAGS_threads;
```
Comment (jowlyzhang) [inline cache/cache_bench_tool.cc:683]: Just for my own learning, why this formula is not just (FLAGS_cache_size * FLAGS_pinned_ratio) / FLAGS_threads?

File: cache/cache_bench_tool.cc [hunk 2]
```diff
@@ -63,6 +63,14 @@ DEFINE_int32(
 DEFINE_uint32(skew, 5, "Degree of skew in key selection. 0 = no skew");
 DEFINE_bool(populate_cache, true, "Populate cache before operations");
 
+DEFINE_double(pinned_ratio, 0.25,
+              "Keep this roughly this portion of entries pinned in cache.");
```
Comment (jowlyzhang) [inline cache/cache_bench_tool.cc:67]: Very nit: first "this" seems to be a redundant.

**Review by jowlyzhang — APPROVED** [review]
Thank you for the PR. LGTM!

File: cache/cache_bench_tool.cc (hunk 1 above)
Comment (pdillinger) [reply cache/cache_bench_tool.cc]: Modified slightly to make the calculation more clear. Thanks

**Pull request closed without merging**